calyx-py/calyx/builder.py

```diff
@@ -1092,14 +1092,24 @@ def invoke(cell: CellBuilder, **kwargs) -> ast.Invoke:
     The keyword arguments should have the form `in_*`, `out_*`, or `ref_*`, where
     `*` is the name of an input port, output port, or ref cell on the invoked cell.
     """
+
+    def try_infer_width(x):
+        width = cell.infer_width(x)
+        if not width:
+            raise WidthInferenceError(
+                "Cannot infer width. "
+                "Consider providing width as an argument."
```

Comment:
Nice, thanks! I wonder if we can make the error message a little more useful, though, by excavating the names of these cells. Don't spend too much time on this, but here are a few options in increasing order of difficulty and niceness:
- "Could not infer width of input during invoke. Consider..."
- "Could not infer width of input when invoking component `adder`. Consider..."
- "Could not infer width of input `a` when invoking component `adder`. Consider..."

Reply:
These make total sense. I believe this last commit should construct error messages like option 3.